src/app/plugins/marketplace.plugin.ts

```diff
@@ -48,39 +48,21 @@ export class MarketPlacePlugin extends Plugin {
     this._replyToUser(message, embed);
   }
 
-  private _handleListMarket(message: IMessage) {
-    message.channel
-      .fetchMessages({ limit: 100 })
-      .then((msgs) => {
-        const itemsForSale: String[] = [];
-
-        //iterate over messages.
-        msgs.forEach((msg) => {
-          const { content } = msg;
-
-          //If content does not contains either element of array then stop.
-          if (
-            ![this._NEW_LISTING_PREFIX, this._NEW_ALIAS_PREFIX].some((el) => content.includes(el))
-          ) {
-            return;
-          }
-
-          let [, item] = content.split('add');
-          if (item?.length) {
-            const user = msg.author;
-            item += '\n' + user; //adds user to end of listing.
-            itemsForSale.push(item);
-          }
-        });
-
-        const embed = new RichEmbed();
-        embed.setTitle('Items For Sale');
-        embed.setColor('#7289da');
-        embed.setDescription(itemsForSale.reverse().join('\n\n'));
-        this._replyToUser(message, embed);
-        message.delete();
-      })
-      .catch((e) => this.container.loggerService.error(e));
+  private async _handleListMarket(message: IMessage) {
+    const itemsForSale = await this._fetchMessages(message, 300).catch((e) =>
+      this.container.loggerService.error(e)
+    );
+
+    if (!itemsForSale) {
+      return;
+    }
+
+    const embed = new RichEmbed();
+    embed.setTitle('Items For Sale');
+    embed.setColor('#7289da');
+    embed.setDescription(itemsForSale.reverse().join('\n\n'));
+    this._replyToUser(message, embed);
+    message.delete();
```

Comment:
does it get deleted instantly?

Reply:
As soon as the bot replies, to the user its almost instant

Reply:
oh, i see, it deletes teh `!market list` ok cool

Reply:
Yeah, deletes the command, and the actual thing the bot sends goes to dms